src/screens/Chat/index.tsx

```diff
@@ -1,28 +1,163 @@
-import { View } from 'react-native';
-import { Surface } from 'react-native-paper';
-import { useNavigation } from '@react-navigation/native';
-import React, { VFC, useLayoutEffect } from 'react';
+import { Surface, HelperText, Text } from 'react-native-paper';
+import { useNavigation, useRoute } from '@react-navigation/native';
+import { KeyboardAvoidingView, Platform, View, FlatList } from 'react-native';
+import React, { VFC, useState, useCallback, useEffect, useLayoutEffect, useRef } from 'react';
+import {
+  runQuery,
+  createQuery,
+  queryMessages,
+  observeMessages,
+  sortByChannelSegment,
+} from '@amityco/ts-sdk';
 
-import { Header } from 'components';
+import { Header, MessageItem, Loading } from 'components';
 
-import { DrawerStackHeaderProps } from 'types';
+import { t } from 'i18n';
+import useAuth from 'hooks/useAuth';
+import getErrorMessage from 'utils/getErrorMessage';
+
+import { DrawerStackHeaderProps, LoadingState } from 'types';
 
 import styles from './styles';
 
+const QUERY_LIMIT = 10;
+
 const ChatScreen: VFC = () => {
+  const flatList = useRef(null);
+  const [messages, setMessages] = useState<Record<string, Amity.Message>>({});
+  const [loading, setLoading] = useState<LoadingState>(LoadingState.NOT_LOADING);
```

Comment:
From this component, you're using only `NOT_LOADING` and `IS_LOADING_MORE` which is definitely a 2 state value. why not result `loading` of the runQuery? is there a blocker?